File: web/src/api/game.api.ts
import { get, post } from './_http'

/** 收藏游戏 */
export const favoriteGame = post('/favorite')
/** 取消收藏游戏 */
export const unfavoriteGame = post('/unfavorite')
/** 获取游戏地址 */
export const queryGameUrl = post('/redirect-to-game')
/** 查询所有游戏 */
export const queryAllGame = get('/get-game-list')
/** 热门游戏 */
export const queryHotGames = post('/hot')

File: victory/lib/apis/game.api.dart
part of 'apis.dart';

class _GameApi {
  final queryGame = http.get('/app/get-game-list', decoder: models.VicGamesModel.fromJson, silent: true);
  final queryGameUrl = http.post('/app/redirect-to-game', loading: true);
  final queryFavorites = http.get('/app/list-favorites', decoder: models.VicFavsModel.fromJson);
  final favorite = http.post('/app/add-favorite', loading: true);
  final unfavorite = http.post('/app/remove-favorite', loading: true);
}
